server/app/controllers/applicant/UpsellController.java

```diff
@@ -165,4 +180,45 @@ public CompletionStage<Result> considerRegister(
               throw new RuntimeException(ex);
             });
   }
+
+  /** Download a PDF file of the application to the program. */
+  @Secure
+  public CompletionStage<Result> download(
```

Comment:
**suggestion (blocking):** Please add new test cases in https://github.com/civiform/civiform/blob/main/server/test/controllers/applicant/UpsellControllerTest.java for this new method including a test case with an authenticated applicant or TI without permission to view the application they're attempting to download.